src/verify-app-guide/verification-steps/application-dependencies.component.js

```diff
@@ -0,0 +1,122 @@
+import React, { useState, useEffect } from "react";
+import CodeOutput from "../../shared/code-output.component";
+import { getAppDependencies } from "../../shared/use-app-shared-deps.hook";
+
+export default React.forwardRef(function ApplicationDependencies(
+  { app, stepNumber },
+  ref
+) {
+  const [isExpanded, setIsExpanded] = useState(false);
+  const [error, setError] = useState(null);
+  const [sharedDependencies, setSharedDependencies] = useState([]);
+  const [missingDeps, setMissingDeps] = useState([]);
+
+  if (ref) {
+    ref.current = {
+      runTest,
+      resetError() {
+        setError(false);
+      },
+    };
+  }
+
+  useEffect(() => {
+    if (missingDeps.length > 0) {
+      setError(true);
+      setIsExpanded(true);
+    }
+  }, [missingDeps]);
+
+  return (
+    <div className="step">
+      <div
+        role="button"
+        tabIndex={0}
+        className="step-header"
+        onClick={toggleExpanded}
+      >
+        <div className="step-number">{stepNumber}</div>
+        <div className="step-summary">
+          Checking if application has shared dependencies
+        </div>
+      </div>
+      {isExpanded && (
+        <div className="step-detail">
+          {sharedDependencies && (
+            <>
+              <div>Your app has the following shared dependencies:</div>
+              <CodeOutput code={sharedDependencies} />
+            </>
+          )}
+          {error && missingDeps.length > 0 ? (
+            <>
+              <div>Found missing dependencies:</div>
+              <CodeOutput code={missingDeps} />
+              <div>
+                Add them to the playground (using import-map-overrides on
```

Comment:
Could we automatically add these for them, via `importMapOverrides.addOverride('name', url)`? https://github.com/joeldenning/import-map-overrides/blob/master/docs/api.md#addoverride

Or is the point here to train them on how to use import-map-overrides? If so, that's fine with me.

Reply:
I think if we automatically add them, the user might miss this chance of learning how to handle dependencies... but if we don't want the user to experiment with `import-map-overrides`, we can add a button to add for them, hiding the import-map-overrides layer.
Having them manually press the button, may give time for them to read what is actually happening
wdyt? maybe I can do this is a next PR, cause this one is kinda bloated already haha

Reply:
Yeah I think teaching them to use import-map-overrides is important. I'm okay with not doing it in this PR or ever